packages/amplify_flutter/lib/categories/amplify_auth_category.dart

```diff
@@ -207,8 +207,10 @@ class AuthCategory {
         : throw _pluginNotAddedException("Auth");
   }
 
-  Future<SignInResult> signInWithWebUI({AuthProvider? provider}) {
-    var request = SignInWithWebUIRequest(provider: provider);
+  Future<SignInResult> signInWithWebUI(
```

Comment:
Could you update the category API to match this signature instead: 

Future<SignInResult> signInWithWebUI({AuthProvider? provider, SignInWithWebUIOptions options})

Then within amplify_auth_plugin_interface you could create the SignInWithWebUIOptions class. 

Then within amplify_auth_cognito, you could create CognitoSignInWIthWebUIOptions which would have the field "isPreferPrivateSession". 

We discussed internally about this best long term approach for this change, and believe that having this options object would give us the greatest long term flexibility while also avoiding cognito specific and platform specific implementations from leaking into the auth category file.

Reply:
Thanks @fjnoyp for your feedback.

I will modify the `amplify_auth_category.dart` to have `SignInWithWebUIOptions` options. It makes sense to me too.

I saw within `amplify_auth_plugin_interface.dart` there is a class called `SignInWithWebUIRequest`. If I created `SignInWithWebUIOptions`, so what is the key different between "Request" and "Options", I think design this way will lead to confusion later when we adding more params to support new features.

Can you share more thoughts on this?

Reply:
Hi.  Thanks!

The key difference is that "Request" is internal to our library and not exposed in the public API.  While "Options" would be a method parameter in the public API, `amplify_auth_category`

Reply:
Thanks. I have updated the code. Can you help to check